src/store/consts/utils.ts

```diff
@@ -494,96 +490,80 @@ export const getY = (
   amount: BN,
   currentPrice: BN
-): Promise<{
-  amountOut: BN
-  simulateSuccess: boolean
-  poolIndex: number
-}> => {
-  try {
-    const marketProgram = getMarketProgramSync()
-
-    let simulateSuccess: boolean = false
-    let poolIndex: number = 0
-    let i: number = 0
-    const poolIndexes: number[] = []
-    const swapPool = PAIRS[networkType].filter(
-      pool =>
-        (fromToken.equals(pool.tokenX) && toToken.equals(pool.tokenY)) ||
-        (fromToken.equals(pool.tokenY) && toToken.equals(pool.tokenX))
-    )
-    // trunk-ignore(eslint/array-callback-return)
-    PAIRS[networkType].map((pair, index) => {
-      if (
-        (fromToken.equals(pair.tokenX) && toToken.equals(pair.tokenY)) ||
-        (fromToken.equals(pair.tokenY) && toToken.equals(pair.tokenX))
-      ) {
-        poolIndexes.push(index)
-      }
-    })
-    if (!swapPool) {
-      return { amountOut: new BN(0), simulateSuccess: false, poolIndex: 0 }
-    }
+): Promise<{ amountOut: BN; poolIndex: number; simulateSuccess: boolean }> => {
+  const marketProgram = getMarketProgramSync()
+  const filteredPools = findPairs(fromToken, toToken, pools)
+  let swapSimulateRouterAmount: BN = new BN(0)
+  let poolIndex: number = 0
 
-    let swapSimulateRouterAmount: BN = new BN(0)
-    for (const pool of swapPool) {
-      const isXtoY = fromToken.equals(pool.tokenX) && toToken.equals(pool.tokenY)
-      const tickMap = await marketProgram.getTickmap(
-        new Pair(pool.tokenX, pool.tokenY, pool.feeTier)
-      )
+  for (const pool of filteredPools) {
+    const isXtoY = fromToken.equals(pool.tokenX) && toToken.equals(pool.tokenY)
 
-      const ticks: Map<number, Tick> = new Map<number, Tick>()
-      if (ticks.size === 0) {
-        for (const tick of poolTicks[i]) {
-          ticks.set(tick.index, tick)
-        }
-      }
-      if (amount.gt(new BN(0))) {
-        const simulateObject: SimulateSwapInterface = {
-          pair: new Pair(pool.tokenX, pool.tokenY, pool.feeTier),
-          xToY: isXtoY,
-          byAmountIn: true,
-          swapAmount: amount,
-          currentPrice: { v: currentPrice },
-          slippage: slippage,
-          pool: pools[poolIndexes[i]],
-          ticks: ticks,
-          tickmap: tickMap,
-          market: marketProgram
-        }
-        try {
-          const swapSimulateResault = simulateSwap(simulateObject)
-          if (swapSimulateRouterAmount.lt(swapSimulateResault.accumulatedAmountOut)) {
-            swapSimulateRouterAmount = swapSimulateResault.accumulatedAmountOut
-            poolIndex = poolIndexes[i]
-          }
-        } catch (error) {
-          i++
-          continue
-        }
-        simulateSuccess = true
-      } else {
-        swapSimulateRouterAmount = new BN(0)
+    const tickMap = await marketProgram.getTickmap(
+      new Pair(pool.tokenX, pool.tokenY, { fee: pool.fee.v })
+    )
+
+    const ticks: Map<number, Tick> = new Map<number, Tick>()
+    if (ticks.size === 0) {
```

Comment:
i think if is not necessary here since it's the moment after we initialized new 'ticks' map